tfe/data_source_github_app_installation.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package tfe
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTFEGHAInstallation() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGHAInstallationRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"installation_id": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				AtLeastOneOf: []string{"name", "installation_id"},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```

Comment:
When it comes to fetching by name, is there any guarantee of uniqueness in Terraform Cloud? Or do we return the first one that matches?

Reply:
The names are unique because it stores the values of the github app's name as is in TFC. The logic to fetch by name returns the first match, but it is guaranteed to be unique.